deltachat-ios/Helper/NotificationManager.swift

```diff
@@ -33,6 +33,12 @@ public class NotificationManager {
                    let chatId = ui["chat_id"] as? Int,
                    let messageId = ui["message_id"] as? Int,
                    !UserDefaults.standard.bool(forKey: "notifications_disabled") {
+                    
+                    if let lastChatId = AppStateRestorer.shared.restoreLastActiveChatId(),
+                       lastChatId == chatId {
+                        return
+                    }
+
                     let array = DcContext.shared.getFreshMessages()
                     UIApplication.shared.applicationIconBadgeNumber = array.count
```

Comment:
that line can be removed i'd say, `applicationIconBadgeNumber` is set in the following line.